roles/openshift_hosted/tasks/registry/storage/s3.yml

```diff
@@ -21,13 +21,45 @@
       openshift_hosted_registry_storage_s3_cloudfront_keypairid and
       openshift_hosted_registry_storage_s3_cloudfront_privatekeyfile are required
 
+
 # Copy the cloudfront.pem to the host if the baseurl is given
-- name: Copy cloudfront.pem to the registry
-  copy:
-    src: "{{ openshift_hosted_registry_storage_s3_cloudfront_privatekeyfile }}"
-    dest: /etc/s3-cloudfront/cloudfront.pem
-    backup: true
-    owner: root
-    group: root
-    mode: 0600
+- block:
+
+    - set_fact:
+        s3_cloudfront_config_dirname: "/etc/origin"
+        s3_cloudfront_target_volume: "cloudfront-vol"
+
+    - set_fact:
+        s3_cloudfront_remote_privkey_path: "{{ s3_cloudfront_config_dirname }}/{{ openshift_hosted_registry_storage_s3_cloudfront_privatekeyfile | basename }}"
+
+    - name: Ensure s3-cloudfront directory exists
+      file:
+        path: "{{ s3_cloudfront_config_dirname }}"
+        state: directory
+        owner: root
+        group: root
+
+    - name: Copy cloudfront.pem to the registry
+      copy:
+        src: "{{ openshift_hosted_registry_storage_s3_cloudfront_privatekeyfile }}"
+        dest: "{{ s3_cloudfront_remote_privkey_path }}"
+        backup: true
+        owner: root
+        group: root
+        mode: 0600
+
+    - name: Create registry secrets for cloudfront private key file
+      oc_secret:
+        state: present
+        namespace: openshift-infra
+        name: cloudfront
+        files:
+          - name: cloudfront
+            path: "{{ s3_cloudfront_remote_privkey_path }}"
```

Comment:
Instead of creating the directory and coping the file over to the remote host, I think it might be better to use the contents parameter for oc_secret: https://github.com/openshift/openshift-ansible/blob/master/roles/lib_openshift/library/oc_secret.py#L144-L160

I think something like this would work:
```
- name: Create registry secret for cloudfront
  oc_secret:
    state: present
    namespace: openshift-infra  # is this namespace right? I think the registry is deployed to the default namespace currently
    name: docker-registry-cloudfront  # probably better to be more descriptive here
    contents:
      path: cloudfront.pem  # this isn't the full file path, just the relative file path from the mounted secret
      data: "{{ lookup('file', openshift_hosted_registry_storage_s3_cloudfront_privatekeyfile) }}"

- name: Add cloudfront secret to the registry deployment config
  command: > 
    oc volume dc/docker-registry --add --name=cloudfront-vol
    -m /etc/origin --type=secret --secret-name=docker-registry-cloudfront
  register: cloudfront_vol_attach
  failed_when: "'already exists' not in cloudfront_vol_attach.stderr and cloudfront_vol_attach.rc != 0"
```

Some other notes:
- I wonder if it is possible to use a list of conditions for `failed_when`, similar to `when`
- If the cloudfront.pem filename is not hardcoded, then you might want to use the basename filter, but I don't see a reason why we wouldn't just hardcode it, since it only exists in the registry config map and secret definitions.
- the set_fact tasks will no longer be needed (but if they were, I would suggest they be moved to role vars or the vars attribute of the block).
- No need to create the remote directory or copy the keyfile to the remote host